docs/common/craft-parts/reference/plugins/uv_plugin.rst

```diff
@@ -0,0 +1,100 @@
+.. _craft_parts_uv_plugin:
+
+uv plugin
+=========
+
+The uv plugin can be used for Python projects that use the uv build tool.
+
+Keywords
+--------
+
+A sequence of subsections specifying the keywords provided by the plugin and
+describe how the plugin uses them using the same format for keywords used in
+the :ref:`part_properties` reference.
+
+In addition to the common :ref:`plugin <part-properties-plugin>` and
+:ref:`sources <part-properties-sources>` keywords, this plugin provides the
+following plugin-specific keywords:
+
+uv-extras
+~~~~~~~~~
+**Type:** list of strings
+
+Extra dependency groups to build with. Each element of the list is passed
+exactly as ``--extra GROUP``.
+
+Environment variables
+---------------------
+
+Along with the variables defined by the :ref:`Python plugin <craft_parts_python_plugin-environment_variables>`,
+this plugin defines additional variables defined below.
+
+.. note::
+
+    This section describes how this plugin uses ``uv``-specific variables. For
+    more information, as well as a complete list of environment variables
+    for ``uv``, see the `uv documentation <https://docs.astral.sh/uv/configuration/environment/>`__.
+
+UV_FROZEN
+~~~~~~~~~
+**Default value:** true
+
+Whether or not to update the :file:`uv.lock` file. If true, :file:`uv.lock`
+must exist and will be used as the single source of truth for dependency
+versions, with no attempt made to update them before installation.
+
+UV_PROJECT_ENVIRONMENT
+~~~~~~~~~~~~~~~~~~~~~~
+**Default value:** See below
+
+A filepath to the Python virtual environment to build with. By default, this
+variable populates itself with the directory that this plugin will create the
+virtual environment.
+
+UV_PYTHON_DOWNLOADS
+~~~~~~~~~~~~~~~~~~~
+**Default value:** never
+
+Whether or not to automatically download Python if the requested version is
+missing.
```

Comment:
Since the default is a non-standard enum `never`, should we describe how to turn it on? Is the complement `always`? This is the first time I'm running into this in our projects, so if this is an existing pattern, ignore me.

Reply:
It's a string, so maybe put it in quotes to indicate that?